src/js/helpers/time-remaining.js

```diff
@@ -0,0 +1,46 @@
+const TIME_IS_UP = 'Time is up!';
+
+const formatTimeRemaining = (timeInSeconds) => {
+	const minutes = Math.floor(timeInSeconds / 60);
+	const seconds = Math.floor(timeInSeconds) % 60;
+
+	return `${minutes < 10 ? `0${minutes}` : minutes}:${seconds < 10 ? `0${seconds}` : seconds} remaining`;
+};
+
+const getTimeRemaining = () => {
+	const endTimeString = sessionStorage.getItem('endTime');
+	if (!endTimeString) {
+		return null;
+	}
+
+	const endTime = Number.parseInt(endTimeString, 10);
+	const now = new Date().getTime();
+
+	const timeRemainingInSeconds = (endTime - now) / 1000;
+	if (timeRemainingInSeconds < 0) {
+		return TIME_IS_UP;
+	}
+
+	return formatTimeRemaining(timeRemainingInSeconds);
+};
+
+const updateTimeRemaining = () => {
+	const timeRemaining = getTimeRemaining();
+	const timeRemainingElement = document.querySelector('[data-time-remaining]');
+	if (timeRemainingElement) {
+		timeRemainingElement.textContent = timeRemaining;
+
+		if (timeRemaining !== TIME_IS_UP) {
+			requestAnimationFrame(updateTimeRemaining);
```

Comment:
Big fan of seeing a timer not being handled by old `setTimeout()`.

If I'm understanding the MDN page for this function correctly, the reason that we don't need to worry about the timer going faster on high refresh-rate screens is because we're using the value for `endTime` that we're retreiving from `sessionStorage` in `getTimeRemaining`? And that the timer isn't exactly updating every second, but every time we request an animation frame?